pkg/operator/controller/dns/controller_test.go

```diff
@@ -0,0 +1,62 @@
+package dns
+
+import (
+	"github.com/google/go-cmp/cmp"
+	configv1 "github.com/openshift/api/config/v1"
+	operatorv1 "github.com/openshift/api/operator/v1"
+	iov1 "github.com/openshift/cluster-ingress-operator/pkg/api/v1"
+	"github.com/openshift/cluster-ingress-operator/pkg/dns"
+	"testing"
+)
+
+func TestPublishRecordToZones(t *testing.T) {
+	var tests = []struct {
+		name   string
+		zones  []configv1.DNSZone
+		expect []string
+	}{
+		{
+			name: "testing for a successful update of 1 record",
+			zones: []configv1.DNSZone{{
+				ID: "/subscriptions/E540B02D-5CCE-4D47-A13B-EB05A19D696E/resourceGroups/test-rg/providers/Microsoft.Network/dnszones/dnszone.io"},
+			},
+			expect: []string{string(operatorv1.ConditionFalse)},
+		},
+		{
+			name:   "when no zones available",
+			zones:  []configv1.DNSZone{},
+			expect: nil,
+		},
+		{
+			name: "when one zone available and one not available",
+			zones: []configv1.DNSZone{
+				{ID: "/subscriptions/E540B02D-5CCE-4D47-A13B-EB05A19D696E/resourceGroups/test-rg/providers/Microsoft.Network/dnszones/dnszone.io"},
+				{ID: ""}},
+			expect: []string{string(operatorv1.ConditionFalse), string(operatorv1.ConditionTrue)},
+		},
+	}
+
+	for _, test := range tests {
+		var conditions []string
```

Comment:
This can be moved above the `for` loop below that uses it.

Reply:
Actually if I do that it will fail like below ->

[miheer@miheer cluster-ingress-operator]$ GO111MODULE=on go test -v -run  TestPublishRecordToZones ./pkg/operator/controller/dns/...
2019-12-21T14:05:41.854+1000	INFO	operator	log/log.go:26	started zapr logger
=== RUN   TestPublishRecordToZones
--- FAIL: TestPublishRecordToZones (0.00s)
    controller_test.go:61: "when no zones available": expected:
        []string(nil)
        got:
        []string{"False"}
FAIL
FAIL	github.com/openshift/cluster-ingress-operator/pkg/operator/controller/dns	0.010s
[miheer@miheer cluster-ingress-operator]$ vi pkg/operator/controller/dns/controller_test.go

Reply:
It works for me.  Were you starting from the correct revision?  This is the change I am suggesting, which I tested locally:
```diff
@@ -37,7 +37,6 @@ func TestPublishRecordToZones(t *testing.T) {
 	}
 
 	for _, test := range tests {
-		var conditions []string
 		record := &iov1.DNSRecord{
 			Spec: iov1.DNSRecordSpec{
 				DNSName:    "subdomain.dnszone.io.",
@@ -50,6 +49,7 @@ func TestPublishRecordToZones(t *testing.T) {
 			dnsProvider: &dns.FakeProvider{},
 		}
 		actual, _ := r.publishRecordToZones(test.zones, record)
+		var conditions []string
 		for _, dnsStatus := range actual {
 			for _, condition := range dnsStatus.Conditions {
 				conditions = append(conditions, condition.Status)
```